api/docs/http_api_index.md

```diff
@@ -1,13 +1,43 @@
 V1 API Services
 ---------------
 
+The V1 API has been removed. All previous routes can now be found under the
+[Master V3](#Master-V3-HTTP-API) API or the [CA V1](#CA-V1-HTTP-API).
+
+V2 HTTP API
+-----------
+The V2 API has been removed. All previous routes can now be found under the
+[Master V3](#V3-HTTP-API).
+
+Master and CA APIs
+------------------
+
+Beginning with Puppet 4, puppet's HTTP API has been split into two separate
+APIs which are versioned separately. There is now one API for the master and
+one for the certificate authority (CA).
+
+All master endpoints are prefixed with `/puppet`, while all CA endpoints are
+prefixed with `/puppet-ca`. All endpoints are explicitly versioned.
+
+Authorization for these endpoints is still controlled with the `auth.conf`
+authorization system in puppet. When specifying the authorization in
+`auth.conf` the prefix (either `/puppet` or `/puppet-ca`) and the version
+mumber on the paths must be retained; the full request path is used.
+
+Master V3 HTTP API
+------------------
+
 Puppet Agents use various network services which the Puppet Master provides in
 order to manage systems. Other systems can access these services in order to
 put the information that the Puppet Master has to use.
 
-The V1 API is all based off of dispatching to puppet's internal "indirector"
-framework. Every HTTP endpoint in V1 follows the form
-`/:environment/:indirection/:key`, where
+The V3 API contains endpoints of two types: those based off of dispatching to
+puppet's internal "indirector" framework and those that are not (namely the
+[environments endpoint](#Environments-Endpoint)).
+
+Every HTTP endpoint that dispatches to the indirector follows the form:
+`/puppet/v3/:indirection/:key?environent=:environment` where
+
   * `:environment` is the name of the environment that should be in effect for
     the request. Not all endpoints need an environment, but the path component
     must always be specified.
```

Comment:
Okay that makes sense. I'm all for reducing work on the indirector :)

And, yeah, I'd think we don't need to mention puppet server in *this* set of docs, at least for puppet 4.